### PR TITLE
Improve error handling when executing commands

### PR DESCRIPTION
**Pull request by Songmu**
If an exit code can be retrieved from the error, take it.

### CODE REVIEW

**Comment by hayajo** [comment]
LGTM

File: cmdutil/testdata/stubcmd.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+	"INT":  os.Interrupt,
+	"TERM": syscall.SIGTERM,
+}
+
+func main() {
+	var (
+		trap     = flag.String("trap", "", "signals")
+		exit     = flag.Int("exit", 0, "exit status")
+		trapExit = flag.Int("trap-exit", 0, "exit status when trapping signal")
+		sleep    = flag.Float64("sleep", 0, "sleep seconds")
```
Comment (mattn) [inline cmdutil/testdata/stubcmd.go]: why not int64?

File: cmdutil/testdata/stubcmd.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+			for _ = range c {
+				fmt.Println("signal received")
+				if *trapExit > 0 {
+					os.Exit(*trapExit)
+				}
+			}
+		}()
+	}
+	if *sleep > 0 {
+		time.Sleep(time.Duration(float64(time.Second) * *sleep))
```
Comment (mattn) [inline cmdutil/testdata/stubcmd.go]: ditto

File: cmdutil/testdata/stubcmd.go (hunk 1 above)
Comment (Songmu) [reply cmdutil/testdata/stubcmd.go]: Because, I wanted to specify sleep seconds as a decimal number like `-sleep 0.3`.

Comment (mattn) [reply cmdutil/testdata/stubcmd.go]: I prefer to use `flag.Duration` in this case. This is easy to specify seconds (-sleep 3s) or hours/minutes (-sleep 2h45m). But this is problem that is likes and dislikes. And this is just nit-picker. :)

Comment (Songmu) [reply cmdutil/testdata/stubcmd.go]: Oh, I did not know that `flag.Duration` could take `ms`. I Changed to use flag.Duration.

**Pull request merged**